Experiment preselected tabs

im2nguyen opened the pull request:
## 🔗 Relevant links

<!--
Include links to the branch preview, Asana task, and Figma designs wherever possible to make reviewing your PR easier.
When including the preview link, make sure to remove any '.' characters from the branch name:
  - Example: ks.my-branch -> ksmy-branch
-->

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

<!--
Briefly list out the changes proposed in this PR.
-->

## 🤷 Why

<!--
Describe why the changes proposed are needed. Some examples: new feature requested, refactor to make things easier later, styling tweaks requested by design, etc.
-->

## 🛠️ How

<!--
Dive into the approach you took, list resources you referenced, detail other approaches you tried but didn't end up going with, etc.
-->

## 📸 Design Screenshots

<!--
Include a screenshot or two and a link to the designs you referenced with this code. These are both helpful context for reviewers to understand what designs you were looking at when you were putting this code together.
-->

## 🧪 Testing

<!--
Create a checklist for going through how to test your proposed changes. If there is anything to configure before interacting with the project in a browser, such as toggling feature flags, changing machine settings, or simulating behavior in browser dev tools, list those steps first.

- [ ] Step 1
- [ ] Step 2
- [ ] Step 3
- [ ] ...
-->

## 💭 Anything else?

<!--
If there is anything you came across that you chose not to address in this PR but plan to soon, list those items here and any Asana tasks you created to go with them.
-->

ashleemboyer commented:
@im2nguyen did you mean to set the target branch to `amb.spike-preselected-tabs` instead of `main`?

im2nguyen commented:
This addresses some of the issues on the spike doc -- if you use `router.replace` (or even `router.push`) it will re-render the page. `shallow` would will re-render it, it just won't refetch the data

Using the native Javascript windows.history addresses this. The main issue right now is that tabs with huge variances in length will cause the page to jump. even though this is not ideal behavior, this is the existing behavior ([example](https://developer.hashicorp.com/vault/tutorials/secrets-management/pki-engine#step-3-create-a-role)). preselected tabs does NOT introduce this behavior

as a result, i think we should consider shipping this as a "phase 1" approach

ashleemboyer commented:
Just converting to draft as an extra protection to the "do not merge" label, so this isn't accidentally merged while we're discussing things! 💖
